### ISSUE TITLE
[Enhancement] Automatically run a min distribution build when a version increment is merged into OpenSearch core

### ISSUE THREAD

**Issue by cwperks** [opensearch-project/opensearch-build#4225]
### Is your feature request related to a problem? Please describe

The CI checks in the security repo require that the min distribution of core is available on Linux and Ubuntu distros for the respective version. When security received auto-cut version increment PRs, they will always fail until a min distribution of core is available.

Oftentimes, someone will [mention the EE team](https://github.com/opensearch-project/security-dashboards-plugin/pull/1659#issuecomment-1819326494) to ask for a build to be triggered to produce the artifacts. Can there be automation put into place to automatically create a min distro when a version increment is merged into OpenSearch core?

### Describe the solution you'd like

Add automation to kick off a min distribution build after a version increment is merged into OpenSearch core.

### Describe alternatives you've considered

Ping EE team on Github or Slack

**Comment by zelinh** [comment]
[Triage] Right now, when OpenSearch core increments its version, we have a daily workflow to raise the PR for respective version manifest (e.g. https://github.com/opensearch-project/opensearch-build/pull/4113) and add it to our CI Jenkins. 
Once this PR is approved and all checks pass, I remember [this new workflow](https://github.com/opensearch-project/opensearch-build/pull/4064) will help merge it and our distribution build Jenkins job will start to generate OpenSearch core. 
I think the problem is that the version increment PR is cut before the above process so the check fails because the core is not built yet. 
We have a discussion talking about the approach to re-run the CI checks for auto-cut version increment PR for plugins. https://github.com/opensearch-project/opensearch-build/issues/2706, but for now we would need you to re-run the CI checks when the core is built.

**Comment by gaiksaya** [comment]
Since the authentication is switched to GitHub for Jenkins, it is now possible to trigger the workflow as soon as auto-version increment PR gets merged in core repos. No need to wait for cron job or EE team to take care of it. Adding some acceptance criteria in the issue body to take this task to completion.
